packages/tools/client-debugger/client-debugger-view/src/components/client-data-views/ContainerHistoryRecord.tsx

```diff
@@ -0,0 +1,118 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { DefaultPalette, IStackItemStyles, Icon, Stack, StackItem } from "@fluentui/react";
+import React from "react";
+
+import {
+	ConnectionStateChangeLogEntry,
+	ContainerStateChangeKind,
+} from "@fluid-tools/client-debugger";
+
+/**
+ * Input props for {@link _ContainerHistoryView}
+ */
+export interface _ContainerHistoryViewProps {
+	/**
+	 * The connection state history of the container.
+	 */
+	containerHistory: readonly ConnectionStateChangeLogEntry[];
+}
+
+/**
+ * Displays a container's history of connection state changes.
+ */
+export function _ContainerHistoryView(props: _ContainerHistoryViewProps): React.ReactElement {
+	const { containerHistory } = props;
+	const nowTimeStamp = new Date();
+	const historyViews: React.ReactElement[] = [];
+
+	// Newest events are displayed first
+	for (let i = containerHistory.length - 1; i >= 0; i--) {
+		const changeTimeStamp = new Date(containerHistory[i].timestamp);
+		const wasChangeToday = nowTimeStamp.getDate() === changeTimeStamp.getDate();
+
+		const accordionBackgroundColor: IStackItemStyles = {
+			root: {
+				background:
+					containerHistory[i].newState === ContainerStateChangeKind.Connected
+						? "#F0FFF0" // green
+						: containerHistory[i].newState === ContainerStateChangeKind.Attached
+						? "#F0FFFF" // blue
+						: containerHistory[i].newState === ContainerStateChangeKind.Disconnected
+						? "#FDF5E6" // yellow
+						: containerHistory[i].newState === ContainerStateChangeKind.Closed
+						? "#FFF0F5" // red
+						: containerHistory[i].newState === ContainerStateChangeKind.Disposed
+						? "#FFE4E1" // dark red
+						: "#C0C0C0", // grey for unknown state
+				borderStyle: "solid",
+				borderWidth: 1,
+				borderColor: DefaultPalette.neutralTertiary,
+				padding: 3,
+			},
+		};
+
+		const iconStyle: IStackItemStyles = {
+			root: {
+				padding: 10,
+			},
+		};
+
+		historyViews.push(
+			<div>
```

Comment:
Nit: React generally wants the root element of each entry in a rendered list to specify a unique `key` property. This allows it to invalidate changes to the array correctly when content / ordering change. You should be able to generate a unique key string from the metadata in the audience change entry.

If you check the console output while running the test app, you will likely see React logging warnings about this.

All that said, in this case we are completely recreating the array each time values change, so it isn't strictly a bug, just a React best practice, so this isn't critical.

Reply:
Added key here. Also fixed `AudienceView.tsx` for same issue.